staging/src/k8s.io/cli-runtime/pkg/genericclioptions/resource/builder.go

```diff
@@ -130,9 +130,28 @@ func IsUsageError(err error) bool {
 
 type FilenameOptions struct {
 	Filenames []string
+	Kustomize string
 	Recursive bool
 }
 
+func (o *FilenameOptions) validate() []error {
+	var errs []error
+	if len(o.Filenames) > 0 && len(o.Kustomize) > 0 {
+		errs = append(errs, fmt.Errorf("only one of -f or -k can be specified"))
+	}
+	if len(o.Kustomize) > 0 && o.Recursive {
+		errs = append(errs, fmt.Errorf("-R is not allowed to work with -k"))
```

Comment:
I can't find in this PR any other mention of this caveat? Should this be in the doc for `-f`? (you might have added it before though).

Reply:
Since `-k` is a new flag, I prefer to add it to `-k` usage.

Reply:
Added following message to `-k` usage.
` This flag can't be used together with -f or -R.`

Reply:
How about `The -k flag can't be used with -f or -R`

Reply:
updated the error message here to `the -k flag can't be used with -f or -R`